File: Laboratorio_04/c/calculadora/CalculatorClient.c
#include <stdio.h>
#include <stdlib.h>
#include <string.h>
#include <unistd.h>
#include <arpa/inet.h>

#define PORT 8080
#define BUFFER_SIZE 1024

int main(int argc, char const *argv[]) {
    if (argc != 3) {
        fprintf(stderr, "Uso: %s <num1> <num2>\n", argv[0]);
        return 1;
    }

    int num1 = atoi(argv[1]);
    int num2 = atoi(argv[2]);

    int sock = 0;
    struct sockaddr_in serv_addr;
    char buffer[BUFFER_SIZE] = {0};

    if ((sock = socket(AF_INET, SOCK_STREAM, 0)) < 0) {
        printf("\n Error al crear el socket \n");
        return -1;
    }

    serv_addr.sin_family = AF_INET;
    serv_addr.sin_port = htons(PORT);

    // Convertir direcciones IPv4 e IPv6 de texto a forma binaria
    if (inet_pton(AF_INET, "127.0.0.1", &serv_addr.sin_addr) <= 0) {
        printf("\n Dirección inválida / Dirección no soportada \n");
        return -1;
    }

    if (connect(sock, (struct sockaddr *)&serv_addr, sizeof(serv_addr)) < 0) {
        printf("\n Error de conexión \n");
        return -1;
    }

    // Envía los dos números al servidor
    sprintf(buffer, "%d %d", num1, num2);
    send(sock, buffer, strlen(buffer), 0);

    // Recibe y muestra los resultados de las operaciones del servidor
    read(sock, buffer, BUFFER_SIZE);
    printf("%s", buffer);

    close(sock);
    return 0;
}

File: Laboratorio_04/Python/Calculadora/CalculatorClient.py
import socket
import sys

PORT = 8080
BUFFER_SIZE = 1024

def main():
    if len(sys.argv) != 3:
        print(f"Uso: {sys.argv[0]} <num1> <num2>")
        return 1

    try:
        num1 = int(sys.argv[1])
        num2 = int(sys.argv[2])
    except ValueError:
        print("Los argumentos deben ser números enteros.")
        return 1

    server_address = ('127.0.0.1', PORT)
    buffer = f"{num1} {num2}"

    try:
        # Crear un socket TCP/IP
        sock = socket.socket(socket.AF_INET, socket.SOCK_STREAM)

        # Conectar el socket al puerto donde el servidor está escuchando
        sock.connect(server_address)

        # Enviar los datos
        sock.sendall(buffer.encode())

        # Esperar y recibir la respuesta
        data = sock.recv(BUFFER_SIZE)
        print(data.decode())

    except socket.error as e:
        print(f"Error de socket: {e}")
        return -1

    finally:
        sock.close()

    return 0

if __name__ == "__main__":
    main()
